### PR TITLE
pythonPackages.oauthenticator & dependencies

### PR DESCRIPTION
**Pull request by ixxie**
Packaged pythonPackages.oauthenticator and its dependencies, pythonPackages.mwoauth and pythonPackages.globus-sdk.

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
https://github.com/NixOS/nixpkgs/pull/31871#issuecomment-359171966 needs to be fixed first. When that is done, the additional commits that are in this PR can be merged.

**Comment by ixxie** [comment]
@FRidh - fixed #31871 and rebased this PR.

**Comment by FRidh** [comment]
@ixxie can you rebase, just so the duplicate commits are gone.

**Comment by ixxie** [comment]
@FRidh done!

**Comment by FRidh** [comment]
Please check whether tests are run. In case a package does not provide any tests, we set `doCheck = false;` and include a comment stating the reason.

**Comment by ixxie** [comment]
@FRidh - I double checked and as far as I can tell all these packages run tests. When distinct, I made the test dependencies explicit by stating `buildInputs`. It also seems some dependencies were propagated from other dependencies; I made these explicit as well.

**Comment by FRidh** [comment]
Good. Note that with `buildPython*` test dependencies are better specified using `checkInputs`. In that case they are only included when `doCheck = true`

**Comment by ixxie** [comment]
@FRidh - switched them to `checkInputs` now.

File: pkgs/development/python-modules/pytest-tornado/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, python
```
Comment (aborsu) [inline pkgs/development/python-modules/pytest-tornado/default.nix]: Idem

File: pkgs/development/python-modules/globus-sdk/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, python
```
[the remark this answers is not in the document]
Comment (ixxie) [reply pkgs/development/python-modules/globus-sdk/default.nix]: Done.

File: pkgs/development/python-modules/mwoauth/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, python
```
[the remark this answers is not in the document]
Comment (ixxie) [reply pkgs/development/python-modules/mwoauth/default.nix]: Done.

File: pkgs/development/python-modules/oauthenticator/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    inherit pname version;
+    sha256 = "0rlg63ii7sxj1xad2nx6wk1xgv3a8dm0az0q9g2v6hdv9cnc4b55";
+  };
+
+  checkInputs = [  globus-sdk mwoauth codecov flake8 pytest
+    pytestcov pytest-tornado requests-mock pyjwt ];
+  
+  propagatedBuildInputs = [ jupyterhub ];
+
+  disabled = pythonOlder "3.4";
```
[the remark this answers is not in the document]
Comment (ixxie) [reply pkgs/development/python-modules/oauthenticator/default.nix]: Didn't work unfortunately.

File: pkgs/development/python-modules/pytest-tornado/default.nix (hunk 1 above)
Comment (ixxie) [reply pkgs/development/python-modules/pytest-tornado/default.nix]: Done.

**Comment by FRidh** [comment]
```
Ran 0 tests in 0.000s
```
No tests are found. Either the test runner cannot find tests and needs to be patched, or there simply are no tests in which case the tests need to be disabled. Do include a comment explaining why the tests are disabled.

This applies to each package.

**Comment by ixxie** [comment]
@FRidh - this is why I was baffled and asking questions about how to really check what is going on: if you remove the `checkInputs` on any of the packages you will see that the build will run tests and they will miss inputs, making the build fail. If I add them, the build is successful but indeed as you note it *claims* no tests are run. From my understanding of the packages, the `checkInputs` are indeed necessary for tests but don't need to be propagated; indeed they are sometimes explicitly listed in the `test-requirements.txt`.

File: pkgs/development/python-modules/oauthenticator/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+buildPythonPackage rec {
+  pname = "oauthenticator";
+  version = "0.7.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0rlg63ii7sxj1xad2nx6wk1xgv3a8dm0az0q9g2v6hdv9cnc4b55";
+  };
+
```
Comment (FRidh) [inline pkgs/development/python-modules/oauthenticator/default.nix]:
```
  checkPhase = ''
    py.test oauthenticator/tests
  '';

  # No tests in archive
  doCheck = false;
```
The tests folder does exist on GitHub.

**Comment by ixxie** [comment]
@FRidh  - followed your suggestion with oauthenticator and reused it for globus-sdk, and then disabled testing explicitly for the other packages.

**Comment by ixxie** [comment]
@FRidh whoopsy! Fixed now.

**Pull request merged**